x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/index.tsx

```diff
@@ -60,6 +61,9 @@ export function TransactionDetails() {
     return config;
   }, [transactionName, transactionType, serviceName]);
 
+  const [selectedBucket, setSelectedBucket] = useState(-1);
+  const traceSamples = distributionData.buckets[selectedBucket]?.samples;
```

Comment:
If you calculate `bucketIndex` from `distributionData.buckets` it should not be necessary to have state here.

Reply:
I already changed it in your first comment! Thanks 👍🏻